WordPress/src/test/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModelTest.kt

```diff
@@ -408,6 +454,103 @@ class ReaderPostDetailViewModelTest {
                 verify(readerFetchRelatedPostsUseCase, times(0)).fetchRelatedPosts(readerPost)
             }
 
+    @Test
+    fun `when related post is clicked from non related post details screen, then related post details screen shown`() =
+            test {
+                whenever(readerFetchRelatedPostsUseCase.fetchRelatedPosts(readerPost))
+                        .thenReturn(
+                                FetchRelatedPostsState.Success(
+                                        localRelatedPosts = ReaderSimplePostList(),
+                                        globalRelatedPosts = globalRelatedPosts
+                                )
+                        )
+                val observers = init(isRelatedPost = false)
+
+                viewModel.onRelatedPostsRequested(readerPost)
+                val relatedPost = observers.uiStates.last().globalRelatedPosts?.cards?.first()
+                relatedPost?.onItemClicked?.invoke(relatedPost.postId, relatedPost.blogId, relatedPost.isGlobal)
+
+                assertThat(observers.navigation[0].peekContent()).isInstanceOf(ShowRelatedPostDetails::class.java)
+            }
+
+    @Test
+    fun `when related post is clicked from related post screen, then related post replaced with history`() =
+            test {
+                whenever(readerFetchRelatedPostsUseCase.fetchRelatedPosts(readerPost))
+                        .thenReturn(
+                                FetchRelatedPostsState.Success(
+                                        localRelatedPosts = ReaderSimplePostList(),
+                                        globalRelatedPosts = globalRelatedPosts
+                                )
+                        )
+                val observers = init(isRelatedPost = true)
+
+                viewModel.onRelatedPostsRequested(readerPost)
+                val relatedPost = observers.uiStates.last().globalRelatedPosts?.cards?.first()
+                relatedPost?.onItemClicked?.invoke(relatedPost.postId, relatedPost.blogId, relatedPost.isGlobal)
+
+                assertThat(observers.navigation[0].peekContent())
+                        .isInstanceOf(ReplaceRelatedPostDetailsWithHistory::class.java)
+            }
+
+    @Test
+    fun `when related post is followed, then related post follow button ui is shown`() =
```

Comment:
Suggestion (💡): This comment of mine is pretty similar to the above comment where I suggest extracting this unique parameter that changes the test behavior and make it visible to the reader. In the end these tests can look something like:
```
    @Test
    fun `when related post is followed, then related post follow button ui is shown`() =
            test {
                val isFollowed = true
                isFollowed(isFollowed)
                val uiStates = init().uiStates

                viewModel.onRelatedPostsRequested(readerPost)
                uiStates.last().globalRelatedPosts?.cards?.first()?.followButtonUiState?.onFollowButtonClicked?.invoke()

                assertThat(uiStates.last().globalRelatedPosts?.cards?.first()?.followButtonUiState?.isFollowed)
                        .isEqualTo(isFollowed)
            }

    @Test
    fun `when related post is un-followed, then related post un-follow button ui is shown`() =
            test {
                val isFollowed = false
                isFollowed(isFollowed)
                val uiStates = init().uiStates

                viewModel.onRelatedPostsRequested(readerPost)
                uiStates.last().globalRelatedPosts?.cards?.first()?.followButtonUiState?.onFollowButtonClicked?.invoke()

                assertThat(uiStates.last().globalRelatedPosts?.cards?.first()?.followButtonUiState?.isFollowed)
                        .isEqualTo(isFollowed)
            }
```